Scripts DICOM/writeTROTSToDICOM.py

```diff
@@ -215,6 +224,198 @@
                         rc.ContourSequence.append(cont)
             rds.ROIContourSequence.append(rc)
-            
 
         rds.save_as("./DICOMs/"+patientFolder+'/rtstruct.dcm', write_like_original = False)
+        
+        if((folder == 'Protons') and ('beamlistfolder' in locals()) and ('machinedata' in locals())):
+            # write rtplan
+            print('Working on rtplan...')
+            beamSigmaEnergy = machinedata["BeamInfo"]["EnergyRangeList"]
+            beamSigmas = machinedata["BeamInfo"]["Sigma"]
+            SAD = [machinedata["BeamInfo"]["SAD"][0],machinedata["BeamInfo"]["SAD"][1]]
+            meta = pydicom.Dataset()
+            SOPInstanceUID = pydicom.uid.generate_uid()
+            meta.MediaStorageSOPClassUID = pydicom.uid.RTPlanStorage
+            meta.MediaStorageSOPInstanceUID = SOPInstanceUID
+            meta.TransferSyntaxUID = pydicom.uid.ExplicitVRLittleEndian  
+            
+            rtds = Dataset()
+            rtds.file_meta = meta
+            rtds.SOPClassUID = pydicom.uid.RTPlanStorage
+            rtds.SOPInstanceUID = SOPInstanceUID
+            rtds.StudyDate = rds.StudyDate
+            rtds.SeriesDate = rds.StudyDate
+            rtds.StudyTime = rds.StudyTime
+            rtds.AccessionNumber = rds.AccessionNumber
+            rtds.Modality = "RTPLAN"
+            rtds.Manufacturer = ds.Manufacturer
+            rtds.InstitutionName = ""
+            rtds.ReferringPhysicianName = ""
+            rtds.OperatorsName = ""
+            rtds.ManufacturerModelName    = ""
+            rtds.PatientName = rds.PatientName
+            rtds.PatientID = rds.PatientID
+            rtds.PatientBirthDate = rds.PatientBirthDate
+            rtds.PatientSex = rds.PatientSex
+            rtds.StudyInstanceUID = rds.StudyInstanceUID
+            rtds.SeriesInstanceUID = rds.StudyInstanceUID
+            rtds.StudyID = rds.StudyID
+            rtds.SeriesNumber = 1
+            rtds.InstanceNumber = 1 
+            rds.FrameOfReferenceUID = rds.FrameOfReferenceUID
+            rtds.PositionReferenceIndicator = ''
+            rtds.RTPlanDate = ""
+            rtds.RTPlanTime = ""
+            rtds.RTPlanGeometry = "PATIENT"
+            
+            rtds.PatientSetupSequence = Sequence()
+            patientSetup = Dataset()
+            patientSetup.PatientPosition = mat["patient"]["PatientPosition"]
+            patientSetup.PatientSetupNumber = 1
+            patientSetup.PatientSetupLabel = "Standard"
+            rtds.PatientSetupSequence.append(patientSetup)
+            
+            rtds.IonBeamSequence = Sequence()
+            currentbeamlist = []
+            beaminfo = {}
+            beaminfo["BeamNumber"] = 1
+            beaminfo["FileBeamNumber"] = int(beamlistfolder["BeamList"][int(patientFolder[-2:])-1][0][0])
+            beaminfo["RangeShifter"] = beamlistfolder["BeamList"][int(patientFolder[-2:])-1][0][4]
+            beaminfo["FinalCumulativeMetersetWeight"] = 0

```

Comment:
Beam6 has a wrong value of "FinalCumulativeMetersetWeight=0", please cross-check.